docs/advanced.md

```diff
@@ -617,3 +617,46 @@ try {
 }
 console.info('finish')
 ```
+
+### Using Inquirer to make interactive tools
+
+Using Inquirer together with yargs is a powerful way to make your CLI tools more interactive and responsive to the user.
+
+One example would be to use `input` to ensure the user inputs required arguments.
+
+```js
+import yargs from 'yargs/yargs';
+import { hideBin } from 'yargs/helpers';
+import { input } from '@inquirer/prompts';
+
+yargs(hideBin(process.argv))
+  .command({
+    command: 'login [username]',
+    describe: 'Log in with defined username',
+    builder: {
+      username: {
```

Comment:
This builder object is adding an option called `username` rather than supplying a description and type for the positional `username`. Need to use `.positional()` for configuring positional.

Reply:
You can confirm this by checking the help, e.g.
```
node index.mjs login --help 
```

Reply:
Ah yes I see, didn't realize it was creating an option.

So I guess I have a question, I know there are two ways to create commands, what is the preferred syntax? I'm sure it's just personal preference but just curious.

```js
yargs(hideBin(process.argv))
  .command({
    command: 'login [username]',
    describe: 'Prints out the username',
    builder: (yargs) => {
      yargs.positional('username', {
        describe: 'The username to be printed',
        type: 'string',
      });
    },
    handler: async (argv) => {
      if (!argv.username) {
        argv.username = await input({ message: 'Please enter your username' });
      }

      console.log(`Welcome back, ${argv.username}!`);
    },
  })
  .help()
  .parse();
```
or

```js
yargs(hideBin(process.argv))
  .command(
    'login [username]',
    'Prints out the username',
    (yargs) => {
      yargs.positional('username', {
        describe: 'The username to be printed',
        type: 'string',
      });
    },
    async (argv) => {
      if (!argv.username) {
        argv.username = await input({ message: 'Please enter your username' });
      }

      console.log(`Welcome back, ${argv.username}!`);
    },
  )
  .help()
  .parse();
```

Reply:
Yes, personal preference. For this example I prefer the object style you used so that the setings are named (e.g. `command`, `describe`, `handler`) rather than just multiple strings and functions.